AdminTool internals refactorings

chrisvest opened the pull request:
This makes it slightly easier to test command providers, and also aligns the visibilities.
For instance, methods that returned `Result` objects were publicly visible, while the `Result` interface was private.

chrisvest committed:
AdminTool internals refactorings

This makes it slightly easier to test command providers, and also aligns the visibilities.
For instance, methods that returned `Result` objects were publicly visible, while the `Result` interface was private.

spacecowboy commented:
@chrisvest lgtm

lutovich commented on community/command-line/src/main/java/org/neo4j/commandline/admin/CommandLocator.java:
```diff
@@ -0,0 +1,73 @@
+ * help output.
+ */
+public interface CommandLocator
+        extends Function<String, AdminCommand.Provider>, Supplier<Iterable<AdminCommand.Provider>>
```
Does anyone actually use `CommandLocator` as `Function` or `Supplier`?
This type signature looks really complex and methods lack good naming because of these interfaces.

chrisvest committed:
Fix license headers in the command-line module

chrisvest committed:
Fix license headers in dbms module

chrisvest committed:
Fix NOTICE files in command-line and dbms modules

chrisvest committed:
Make the CommandLocator interface grow up

It no longer just extends two functional interfaces (which ironically made it a non-functional interface).
It now has methods with meaningful names.
It now has javadocs that makes sense.

chrisvest commented:
@spacecowboy @lutovich I think this is good now.
